### PR TITLE
RE2022-79: Add workspace to config / startup

### PR DESCRIPTION
**Pull request by MrCreosote**
Will be needed for matchers.

Tested startup code manually with good & bad workspace urls. The get_workspace method is not yet tested; it'll get tested when it's used.

The http_bearer changes were tested with by printing the user in one of the endpoints.

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -18,9 +18,10 @@ services:
       - KBCOLL_ARANGO_DB=collections_test
       - KBCOLL_ARANGO_USER=root
       - KBCOLL_ARANGO_PWD=foobar
-      - KBASE_AUTH_URL=https://ci.kbase.us/services/auth
+      - KBCOLL_KBASE_AUTH2_URL=https://ci.kbase.us/services/auth
```
Comment (MrCreosote) [inline docker-compose.yaml:21]: oops

File: src/service/app_state.py [hunk 2]
```diff
@@ -58,6 +62,18 @@ async def _build_storage(cfg: CollectionsServiceConfig, data_products: list[Data
         raise
 
 
+async def _get_workspace_url(cfg: CollectionsServiceConfig) -> str:
+    if cfg.dont_connect_to_external_services:
+        return False
+    ws = Workspace(cfg.workspace_url)
```
Comment (Tianhao-Gu) [inline src/service/app_state.py:68]: OK. But just in case, I would move `ws = Workspace(cfg.workspace_url)` to the try-except clause below.

**Review by Tianhao-Gu — DISMISSED** [review]
Overall LGTM. Just have one comment.

File: src/service/app_state.py (hunk 2 above)
Comment (MrCreosote) [reply src/service/app_state.py]: The only error that could occur here is a url parse error, for which the catch / rethrow isn't really necessary as the error message should provide enough info, but it doesn't hurt either, so fair enough

**Comment by MrCreosote** [comment]
The last commit was a trivial change requested by the reviewer and the rest of the code is approved, so I'm going to go ahead and merge. If changes are necessary we can do another PR

**Pull request merged**